From 8ccf8890fed7097becbe9c7fe8ca89741886fe51 Mon Sep 17 00:00:00 2001
From: Rich Megginson <rmeggins@redhat.com>
Date: Thu, 3 Apr 2014 20:50:14 -0600
Subject: [PATCH 193/225] Ticket #47492 - PassSync removes User must change
 password flag on the Windows side

https://fedorahosted.org/389/ticket/47492
Reviewed by: nkinder (Thanks!)
Branch: 389-ds-base-1.2.11
Fix Description: Cherry pick broke outbound sync.  The suffix variable in
map_entry_dn_outbound was initialized to NULL and not set.  This patch
restores the previous functionality removed by the cherry pick.
NOTE: On 1.3.0 and later, the code has changed quite a bit, so that code
is not affected
Platforms tested: RHEL5 x86_64
Flag Day: no
Doc impact: no
(cherry picked from commit ab4b92153f2d0f5fa3afe9ab2978deaf85fe5e5b)
---
 ldap/servers/plugins/replication/windows_protocol_util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ldap/servers/plugins/replication/windows_protocol_util.c b/ldap/servers/plugins/replication/windows_protocol_util.c
index 811d2fd..627e8ad 100644
--- a/ldap/servers/plugins/replication/windows_protocol_util.c
+++ b/ldap/servers/plugins/replication/windows_protocol_util.c
@@ -3542,7 +3542,7 @@ map_entry_dn_outbound(Slapi_Entry *e,
 	char *guid = NULL;
 	Slapi_DN *new_dn = NULL;
 	int is_nt4 = windows_private_get_isnt4(prp->agmt);
-	const char *suffix = NULL;
+	const char *suffix = slapi_sdn_get_dn(windows_private_get_windows_subtree(prp->agmt));
 	Slapi_Entry *remote_entry = NULL;
 
 	if (NULL == e) {
-- 
1.8.1.4

